front/lib/resources/file_resource.ts

```diff
@@ -0,0 +1,228 @@
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+import type {
+  FileId,
+  FileType,
+  LightWorkspaceType,
+  Result,
+} from "@dust-tt/types";
+import { Err, FILE_ID_PREFIX, Ok } from "@dust-tt/types";
+import type {
+  Attributes,
+  CreationAttributes,
+  ModelStatic,
+  Transaction,
+} from "sequelize";
+import type { Readable, Writable } from "stream";
+
+import config from "@app/lib/api/config";
+import type { Authenticator } from "@app/lib/auth";
+import { getPrivateUploadBucket } from "@app/lib/file_storage";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import { FileModel } from "@app/lib/resources/storage/models/files";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+import { generateModelSId } from "@app/lib/utils";
+
+type FileVersion = "processed" | "original";
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export interface FileResource extends ReadonlyAttributesType<FileModel> {}
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export class FileResource extends BaseResource<FileModel> {
+  static model: ModelStatic<FileModel> = FileModel;
+
+  constructor(model: ModelStatic<FileModel>, blob: Attributes<FileModel>) {
+    super(FileModel, blob);
+  }
+
+  static async makeNew(
+    blob: Omit<CreationAttributes<FileModel>, "status" | "sId">
+  ) {
+    const fileId = makeDustFileId();
```

Comment:
let's not introduce new terminology, this is an `sId` right?

Reply:
Fixed